runtime/imports.go

```diff
@@ -485,28 +485,129 @@ func ext_sr25519_generate(context unsafe.Pointer, idData, seed, seedLen, out int
 //export ext_ed25519_public_keys
 func ext_ed25519_public_keys(context unsafe.Pointer, idData, resultLen int32) int32 {
 	log.Trace("[ext_ed25519_public_keys] executing...")
-	log.Warn("[ext_ed25519_public_keys] Not yet implemented.")
-	return 0
+	instanceContext := wasm.IntoInstanceContext(context)
+	memory := instanceContext.Memory().Data()
+
+	mutex.RLock()
+	runtimeCtx := registry[*(*int)(instanceContext.Data())]
+	mutex.RUnlock()
+
+	keys := runtimeCtx.keystore.Ed25519PublicKeys()
+	// TODO: when do deallocate?
```

Comment:
Can you explain?

Reply:
we malloc some space in the memory so we can write the keys, but the memory never gets freed which is bad and could cause us to run out of memory. it needs to be freed at some point, but I'm not sure where to add that in. it should be freed once the runtime call is done, but there's no function that is called when the call finished